### PR TITLE
Enable upstream/downstream dbt deps

### PR DESCRIPTION
**Pull request by clairelin135**
Adds utility methods to `DbtManifest` to enable fetching upstream/downstream dbt asset keys.

**Defining python assets upstream of a dbt asset**
Upstream node of a dbt asset is a single-table source
```python
@asset(
    key=manifest.get_asset_key_for_source("raw_data"),
)
def raw_inventory() -> Any:
    ...
```
Upstream node of a dbt asset is a multi-table source
```python
@multi_asset(
    outs={
        name: AssetOut(key=asset_key)
        for name, asset_key in manifest.get_asset_keys_by_output_name_for_source(
            "clients_data"
        ).items()
    }
)
def clients_data():
    ....
```

**Defining python assets downstream of a dbt asset**
```python
@asset(
    non_argument_deps={
        manifest.get_asset_key_for_model("daily_order_summary").with_prefix("prefix")
    }
)
def downstream_python_asset():
    ...
```

### CODE REVIEW

File: examples/experimental/tutorial_dbt_dagster_v2/tutorial_dbt_dagster_v2/constants.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from dagster import file_relative_path
 
 DBT_PROJECT_DIR = file_relative_path(__file__, "../../jaffle_shop")
-MANIFEST_PATH = file_relative_path(__file__, "../../jaffle_shop/target/manifest.json")
+MANIFEST_PATH = file_relative_path(__file__, "../jaffle_shop/target/manifest.json")
```
Comment (rexledesma) [inline examples/experimental/tutorial_dbt_dagster_v2/tutorial_dbt_dagster_v2/constants.py:4]: Does this need to change?

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py [hunk 2]
```diff
@@ -130,6 +132,95 @@ def get_asset_key_by_output_name(self, output_name: str) -> AssetKey:
...
+                f"Could not find a dbt node with unique_id: {unique_id}. A unique ID consists of"
+                " the node type (model, source, seed, etc.), project name, and node name in a"
+                " dot-separated string. For example: model.my_project.my_model\n For more"
+                " information on the unique ID structure:"
+                " https://docs.getdbt.com/reference/artifacts/manifest-json"
+            )
+
+        return self.node_info_to_asset_key(node_info)
+
+    def get_asset_key_for_source(self, source_name: str) -> AssetKey:
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py:149]: docstring?

**Review by rexledesma — APPROVED** [review]
🙏 Let's announce this in a public discussion so we can get feedback on these API's

**Comment by clairelin135** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14932** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14932" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/14932?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py (hunk 2 above)
Comment (clairelin135) [reply python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py:149]: Updated to include an example

File: examples/experimental/tutorial_dbt_dagster_v2/tutorial_dbt_dagster_v2/constants.py (hunk 1 above)
Comment (rexledesma) [reply examples/experimental/tutorial_dbt_dagster_v2/tutorial_dbt_dagster_v2/constants.py:4]: `DBT_PROJECT_DIR` should change as well then to remove the extra `..`

**Pull request merged**